### PR TITLE
Add tags support for resource_aws_lightsail_instance

### PR DESCRIPTION
**Pull request by blckct**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Add tags support for Lightsail instances.

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource/aws_lightsail_instance.go:  Add `tags` argument
```

Output from acceptance testing:

```
make testacc TEST=./aws TESTARGS='-run=TestAccAWSLightsailInstance_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -parallel 20 -run=TestAccAWSLightsailInstance_basic -timeout 120m
=== RUN   TestAccAWSLightsailInstance_basic
=== PAUSE TestAccAWSLightsailInstance_basic
=== CONT  TestAccAWSLightsailInstance_basic
--- PASS: TestAccAWSLightsailInstance_basic (85.64s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	85.671s

make testacc TEST=./aws TESTARGS='-run=TestDiffLightsailTags'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -parallel 20 -run=TestDiffLightsailTags -timeout 120m
=== RUN   TestDiffLightsailTags
--- PASS: TestDiffLightsailTags (0.00s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	0.018s
...
```

### CODE REVIEW

File: aws/resource_aws_lightsail_instance.go [hunk 1]
```diff
@@ -233,6 +245,19 @@ func resourceAwsLightsailInstanceDelete(d *schema.ResourceData, meta interface{}
 	return nil
 }
 
+func resourceAwsLightsailInstanceUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).lightsailconn
+
+	if d.HasChange("tags") && !d.IsNewResource() {
```
Comment (bflad) [inline aws/resource_aws_lightsail_instance.go]: The resource `Create` function does not call `Update`, so the `d.IsNewResource()` check is extraneous. 👍 

```suggestion
	if d.HasChange("tags") {
```

File: aws/resource_aws_lightsail_instance_test.go [hunk 2]
```diff
@@ -32,6 +32,18 @@ func TestAccAWSLightsailInstance_basic(t *testing.T) {
 					resource.TestCheckResourceAttrSet("aws_lightsail_instance.lightsail_instance_test", "blueprint_id"),
 					resource.TestCheckResourceAttrSet("aws_lightsail_instance.lightsail_instance_test", "bundle_id"),
 					resource.TestCheckResourceAttrSet("aws_lightsail_instance.lightsail_instance_test", "key_pair_name"),
+					resource.TestCheckResourceAttr("aws_lightsail_instance.lightsail_instance_test", "tags.%", "1"),
```
Comment (bflad) [inline aws/resource_aws_lightsail_instance_test.go]: Could you please move the tags acceptance testing to its own acceptance test function and configuration, leaving the original test configuration for the basic test and check this value is 0? Thanks!

```suggestion
					resource.TestCheckResourceAttr("aws_lightsail_instance.lightsail_instance_test", "tags.%", "0"),
```

File: aws/resource_aws_lightsail_instance_test.go [hunk 3]
```diff
@@ -187,6 +199,28 @@ resource "aws_lightsail_instance" "lightsail_instance_test" {
...
+  tags = {
+    Name = "tf-test"
+  }
+}
+`, lightsailName)
+}
+
+func testAccAWSLightsailInstanceConfig_updated(lightsailName string) string {
+	return fmt.Sprintf(`
+provider "aws" {
```
Comment (bflad) [inline aws/resource_aws_lightsail_instance_test.go]: Please note that we are trying to remove hardcoded `provider` configurations from the test configurations and there is a pending PR for fixing Lightsail here: https://github.com/terraform-providers/terraform-provider-aws/pull/9080

File: aws/resource_aws_lightsail_instance.go (hunk 1 above)
Comment (blckct) [reply aws/resource_aws_lightsail_instance.go]: Yeah, copy/paste error, sorry.

File: aws/resource_aws_lightsail_instance_test.go (hunk 2 above)
Comment (blckct) [reply aws/resource_aws_lightsail_instance_test.go]: Done.

File: aws/resource_aws_lightsail_instance_test.go (hunk 3 above)
Comment (blckct) [reply aws/resource_aws_lightsail_instance_test.go]: I think I've done it correctly this time but please check.

**Comment by blckct** [comment]
I've commited the code review changes and then squashed everything into one commit. Thanks for reviewing this.

**Comment by blckct** [comment]
Thanks.

**Comment by bflad** [comment]
This has been released in [version 2.19.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2190-july-11-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

For further feature requests or bug reports with this functionality, please create a [new GitHub issue](https://github.com/terraform-providers/terraform-provider-aws/issues/new/choose) following the template for triage. Thanks!

**Pull request merged**